engine/lite/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/ScenarioInterpreterFactory.scala

```diff
@@ -266,6 +267,8 @@ object ScenarioInterpreterFactory {
         case er: EndReference =>
           //FIXME: do we need it at all
           monad.pure[ResultType[PartResult]](Writer.value(irs.map(ir => EndPartResult(er.nodeId, ir.finalContext, null.asInstanceOf[Res]))))
+        case fer: FragmentEndReference =>
+          monad.pure[ResultType[PartResult]](Writer.value(irs.map(ir => EndPartResult(fer.nodeId, ir.finalContext, fer.declaredVariables.asInstanceOf[Res]))))
```

Comment:
I'm just curious, why do you need return this result? For req-resp it is necessary to return it via HTTP response. For tests I think that you need only to notify listener. Don't you?

Reply:
I see that it can be helpful for our scenario testing mechanism (TestScenarioRunner)

Reply:
WDYT by "I couldn't see what we are getting at the end of fragment."? To see where? In testing mechanism. Test results for this ending node are reported before you create `FragmentEndReference`, here: https://github.com/TouK/nussknacker/pull/4492/files/915fa585694829a0c7909f54d9638514244a5ff9#diff-7dcb7c349f9082c18d9b7b2cfd5524059175576524e1e715d4cb6ee52994193bR111

Reply:
It is strongly correlated with conversation we are having bellow. Since I've used under the hood mechanism to report sink output to `externalInvocationResults` two things happend:
- I've looked up `externalInvocationResults` in tests and what I've got in `EndPartResult` I have also got in `externalInvocationResults` result. That is why I've passed this result here. 
- `externalInvocationResults` are also sent to FE, so passing null here crashes NU FE. (`TypeError: Cannot read properties of null (reading 'pretty')`) 

What I think I should do is not to pass result to `EndPartResult` and in the same time do not report sink invocation to `externalInvocationResults`. To do that I would have to remove `collectSinkResults` from `TestRunner.scala`. WDYT?

Reply:
I will close it, since it is connected with conversation we are having bellow.